### DISCUSSION TITLE
Add a new Assembly Option to display or show x86 code style syntax instead of MIPS code style syntax

### DISCUSSION THREAD

**Discussion by ghost** [grumpycoders/pcsx-redux#1100] [Ideas]
This new Assembly Option may be available but only when the "Dynarec CPU" option in the "Emulation Configuration" window or tab is enabled probably.

In No$psx 2.0 PlayStation one emulator and debugger the user can already change "Disassembler Syntax" from "Native MIPS-style" to "Nocash Z80/X86-style" and I would like that PCSX-Redux also has a new Assembly Option to do that.

With this new Assembly Option PCSX-Redux can be more familiar to both MIPS and x86 assembly programmers and not only MIPS programmers! What do you think about this crazy idea!?

**Comment by wheremyfoodat** [comment]
> What do you think about this crazy idea!?

It is feasible, even without the dynarec enabled (as the CPU emulation core and the disassembler are totally independent of each other). However both @nicolasnoble and I are not particularly fond of nocash's x86-like syntax. We both find it to be unintuitive as x86 doesn't always map well to MIPS and pretty confusing. Even though I've spent hundreds of hours more with x86 code than MIPS code, I still find myself always switching no$psx to the MIPS syntax, and most people I know do the same. Plus there's a pretty significant amount of baggage and work to add+maintain a second disassembly mode.

**Reply by ghost** [reply]
> > What do you think about this crazy idea!?
> 
> It is feasible, even without the dynarec enabled (as the CPU emulation core and the disassembler are totally independent of each other). However both @nicolasnoble and I are not particularly fond of nocash's x86-like syntax. We both find it to be unintuitive as x86 doesn't always map well to MIPS and pretty confusing. Even though I've spent hundreds of hours more with x86 code than MIPS code, I still find myself always switching no$psx to the MIPS syntax, and most people I know do the same. Plus there's a pretty significant amount of baggage and work to add+maintain a second disassembly mode.

This feature does**n't** have to be implemented **exactly the same** as how No$psx does and I don't think that adding and implementing this feature can be done in a short time at all but I am glad to hear your opinions though.
